app/controllers/holds_controller.rb

```diff
@@ -47,7 +60,49 @@ def holds_not_ready
       holds.reject(&:ready_for_pickup?)
     end
 
+    def handle_pickup_change_request
+      change_pickup_response = symphony_client.change_pickup_library(
+        hold_key: @hold_to_act_on.key,
+        pickup_library: params['pickup_library'],
+        session_token: current_user.session_token
+      )
+      case change_pickup_response.status
+      when 200
+        process_flash(:success, 'update_pickup.success_html')
+      else
+        Rails.logger.error(change_pickup_response.body)
+        process_flash(:error, 'update_pickup.error_html')
+      end
+    end
+
+    def handle_not_needed_after_request
+      raise HoldException, 'Error' if Date.parse(params['hold_expiration_date']) < Date.today
```

Comment:
👌

Reply:
It's something I caught by mistake really, SUL didn't cover this problem.